File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    st=[]
    for cur in s:
        if cur==')':
            if st and st[-1]=='(':
                st.pop()
            else:
                return False
        else:
            st.append('(')
    if not st:
        return True
    return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<bits/stdc++.h>

using namespace std;
int cnt;
bool solution(string s)
{
    bool answer = true;
    for (auto cur : s)
    {
        if (cur==')')
        {
            if (!cnt)
                return false;
            cnt-=1;
            continue;
        }
        cnt+=1;
    }
    if (cnt>0)
        answer=false;
    return answer;
}